src/audio/buffer.c

```diff
@@ -7,7 +7,7 @@
 
 #include <sof/audio/buffer.h>
 #include <sof/audio/component.h>
-#include <sof/drivers/interrupt.h>
+#include <rtos/interrupt.h>
```

Comment:
This actually seems a bit wrong. buffer.c (and dai.c, copier.c) should have no need to use the interrupt interface directly. Not necessarily for this PR (rtos/interrupt.h is actually better), but we could probably prune the dependencies from audio code. It should really be the XTOS drivers that need these interfaces.

Reply:
agree, but I need to de-tangle before I can remove. This PR and others in progress are the de-tangling.